File: rust/src/Problem2.rs
fn main() {
    fib();
}

fn fib() {
    let mut first: u32 = 1;
    let mut second: u32 = 2;
    let mut sum = first + second;
    let mut sum_of_evens = sum;
    while sum < 4_000_000 {
        sum = first + second;
        if sum % 2 == 0 {
            sum_of_evens += sum;
        }
        first = second;
        second = sum;
    }

    println!("{}", sum_of_evens);
}

File: dart/lib/src/Problem2.dart
import 'Euler.dart';

class Problem2 implements Euler {
  String answer() {
    var first = 1;
    var second = 2;
    var sum = 2;
    while (first + second < 4000000) {
      var temp = second;
      second = first + second;
      first = temp;
      if (second % 2 == 0) {
        print('even ' + second.toString());
        sum += second;
      }
    }
    return sum.toString();
  }
}
